projects.py

```diff
@@ -0,0 +1,146 @@
+import argparse
+import sqlite3
+import sys
+import warnings
+
+import pandas.io.sql as pdsql
+import pymysql as mdb
+
+from tt import *
+
+warnings.filterwarnings('ignore')
+
+
+def table_check(args_dict, db, exists=True):
+    check_table = db.cursor()
+    if args_dict['dbengine'] == 'mysql':
+        check_table.execute('''
+            select count(*) from information_schema.tables where
+                table_name = "{table}";
+        '''.format(table=args_dict['proj_table']))
+    else:
+        check_table.execute('''
+            select count(*) from sqlite_master where type = 'table' and
+                name = "{table}";
+        '''.format(table=args_dict['proj_table']))
+    if exists:
+        if check_table.fetchone()[0] == 1:
+            check_table.close()
+            pass
+        else:
+            sys.exit('STOP! The table {} does not exist.'
+                     .format(args_dict['proj_table']))
+    else:
+        if check_table.fetchone()[0] != 1:
+            check_table.close()
+            pass
+        else:
+            sys.exit('STOP! The table {} already exists.'
+                     .format(args_dict['proj_table']))
+
+
+def create_projects(args_dict, db):
+    table_check(args_dict, db, exists=False)
+    if args_dict['dbengine']=='mysql':
+        sql_commands = [('''
+            use {db};
+            create table if not exists {table} (
+                id int not null auto_increment,
+                project_description varchar(128),
+                project_code varchar(8),
+                project_bill_nbr varchar(10),
+                primary key (id)
+            );
+        '''.format(db=args_dict['db'], table=args_dict['proj_table']))]
+    elif args_dict['dbengine'] == 'sqlite':
+        sql_commands = [('''
+            create table if not exists {table} (
+                id integer primary key autoincrement,
+                project_description varchar(128),
+                project_code varchar(8),
+                project_bill_nbr varchar(10)
+            );
+        '''.format(table=args_dict['proj_table']))]
+    else:
+        sys.exit(DB_ERROR_MESSAGE)
+
+    if args_dict['dbengine'] == 'mysql':
+        sql_commands.insert(0, 'create database if not exists {};'.format(args_dict['db']))
+
+    sql_commands.append(('''
+        insert into {table} (project_description, project_code, project_bill_nbr)
+            values ('MyFirstProject', 'proj1', '111111');
+    '''.format(table=args_dict['proj_table'])))
+
+    return sql_commands
+
+
+def add_projects(args_dict, db):
```

Comment:
It'd be useful to check if the entry exists before adding to, to avoid entering the same project multiple times.